sdks/rust/src/lib.rs

```diff
@@ -295,6 +312,64 @@ impl TransactionContext for SabreTransactionContext {
             Ok(result_vec)
         }
     }
+
+    fn add_event(
+        &self,
+        event_type: String,
+        attributes: Vec<(String, String)>,
+        data: &[u8],
+    ) -> Result<(), WasmSdkError> {
+        unsafe {
+            let event_type_buffer = WasmBuffer::new(&event_type.as_bytes())?;
+
+            if data.is_empty() {
```

Comment:
Data may be empty - this is for arbitrary application data that requires serialization beyond key-value pairs provided in the attribute section.